rust/lance/src/dataset/scanner.rs

```diff
@@ -546,6 +551,29 @@ impl Scanner {
         }
     }
 
+    fn project_additional<S: AsRef<str>>(
```

Comment:
I'm a little confused by what this method does. Could you document a little more. Seems like it means, "project `columns`, but exclude anything in `base_schema`", is that right?

Reply:
Yes, sorry, will add a comment.  Maybe I can rename it too.  I think you got it right.  It's "given a list of names, figure out which ones are in base schema already, and create a schema that's just the new columns (or none if all the given names were in base schema)"

Reply:
I've renamed this to `calc_new_fields` and added a comment